sensor/kubernetes/listener/resource_event_handler.go

```diff
@@ -76,9 +81,27 @@ func (k *listenerImpl) handleAllEvents() {
 	var crdSharedInformerFactory dynamicinformer.DynamicSharedInformerFactory
 	var complianceResultInformer, complianceProfileInformer, complianceTailoredProfileInformer, complianceScanSettingBindingsInformer, complianceRuleInformer, complianceScanInformer, complianceSuiteInformer cache.SharedIndexInformer
 	var profileLister cache.GenericLister
-	if resourceList, err := serverResourcesForGroup(k.client, complianceoperator.GetGroupVersion().String()); err != nil {
-		log.Errorf("Checking API resources for group %q: %v", complianceoperator.GetGroupVersion().String(), err)
-	} else if resourceExists(resourceList, complianceoperator.ComplianceCheckResult.Name) {
+	crdWatcher := crd.NewCRDWatcher(&k.stopSig, dynamicinformer.NewDynamicSharedInformerFactory(k.client.Dynamic(), noResyncPeriod))
+	coAvailabilityChecker := complianceOperatorAvailabilityChecker.NewComplianceOperatorAvailabilityChecker()
+	if err := coAvailabilityChecker.AppendToCRDWatcher(crdWatcher); err != nil {
+		log.Errorf("Unable to add the Resource to the CRD Watcher: %v", err)
+	}
+	if err := crdWatcher.Watch(k.crdWatcherStatusC); err != nil {
+		log.Errorf("Failed to start watching the CRDs: %v", err)
+	}
+	crdHandlerFn := func(status *watcher.Status) {
+		if status.Available {
+			log.Infof("Resources %v became available", status.Resources)
+			if err := k.pubSub.Publish(&internalmessage.SensorInternalMessage{
+				Kind:     internalmessage.SensorMessageSoftRestart,
+				Text:     "The resources for the Compliance Operator have been detected in the cluster. Sensor will restart the connection to reconcile resources with Central.",
```

Comment:
[nit] suggestion for brevity and readability  
```suggestion
				Text:     "Compliance Operator resources have been updated. Connection will restart to force reconciliation with Central.",
```

Reply:
done